thread_ID, Id, ParentId, PostType, Score, CreationDate, OwnerID, Body
53119129,53119129,null,1,1,Fri Nov 02 13:04:00 EDT 2018,5114014,"<p>I am trying to setup AWS IoT using a Python script as mentioned in this <a href="https://www.hackster.io/awshome/awshome-home-automation-using-rpi-alexa-iot-a3d3dc#toc-installing-dependencies-4" rel="nofollow noreferrer">link</a>:</p>		<p>I am able to connect on AWS IoT MQTT without WebSocket (used x.509 certificate).</p>		<pre><code># creates the AWS IoT 	def createIoT(): 	   iot = AWSIoTMQTTShadowClient('AWSHome') 	   # update this with your own endpoint from the IOT dashboard 	   iot.configureEndpoint('allj.iot.reg.amazonaws.com', 443) 	   iot.configureCredentials('rootCA','private.key','certificate.crt') 	   iot.configureConnectDisconnectTimeout(10)  # 10 sec 	   iot.configureMQTTOperationTimeout(5)  # 5 sec 	   iot.connect() 	   return 	</code></pre>		<p>But When I try to connect on AWS IoT MQTT with WebSocket, I get the following error:</p>		<p>Using the certificate generated by running this command: <a href="https://www.symantec.com/content/en/us/enterprise/verisign/roots/VeriSign-Class%203-Public-Primary-Certification-Authority-G5.pem" rel="nofollow noreferrer">wget</a></p>		<pre><code># creates the AWS IoT 	def createIoT(): 	   iot = AWSIoTMQTTShadowClient('AWSHome') 	   # update this with your own endpoint from the IOT dashboard 	   iot.configureEndpoint('asdasd.reg.amazonaws.com', 443) 	   iot.configureCredentials('VeriSign-Class%203-Public-Primary-Certification-Authority-G5.pem') 	   iot.configureConnectDisconnectTimeout(10)  # 10 sec 	   iot.configureMQTTOperationTimeout(5)  # 5 sec 	   iot.connect() 	   return 	</code></pre>		<p>Error:</p>		<pre><code>    Traceback (most recent call last): 	    File "./awshome.py", line 60, in &lt;module&gt; iot = createIoT() 	    File "./awshome.py", line 50, in createIoT iot.connect() File "/home/pi/.local/lib/python2.7/site-packages/AWSIoTPythonSDK/MQTTLib.py", line 1216, in connect return         	    self._AWSIoTMQTTClient.connect(keepAliveIntervalSecond) 	    File "/home/pi/.local/lib/python2.7/site-packages/AWSIoTPythonSDK/MQTTLib.py", line 485, in connect return self._mqtt_core.connect(keepAliveIntervalSecond) 	    File "/home/pi/.local/lib/python2.7/site-packages/AWSIoTPythonSDK/core/protocol/mqtt_core.py", line 192, in connect self.connect_async(keep_alive_sec, self._create_blocking_ack_callback(event)) 	    File "/home/pi/.local/lib/python2.7/site-packages/AWSIoTPythonSDK/core/protocol/mqtt_core.py", line 219, in connect_async 	    raise e 	    ssl.SSLError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:661)ssl.SSLError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:661)	</code></pre>	"
53119129,53119254,53119129,2,0,Fri Nov 02 13:13:00 EDT 2018,6103001,"<p>It can be that you have self signed certificate, or something else happens that makes the certificate not valid. Question is what do you want to achieve... If the point is to see it working:</p>		<pre><code>import ssl		ssl._create_default_https_context = ssl._create_unverified_context	</code></pre>		<p>That is pretty ugly solution, <a href="https://stackoverflow.com/questions/27835619/urllib-and-ssl-certificate-verify-failed-error">here</a>, and <a href="http://blog.pengyifan.com/how-to-fix-python-ssl-certificate_verify_failed/" rel="nofollow noreferrer">here</a> you have wider explanation.</p>	"
53119129,53119258,53119129,2,1,Fri Nov 02 13:13:00 EDT 2018,2531132,"<p>Since May 2018 both the endpoint and the certificates for AWS's IoT Core has changed. </p>		<p>In short you need to:</p>		<ol>	<li>Change your endpoint to <code>a1am7bjirugllj-ats.iot.us-east-1.amazonaws.com</code> (note the <code>-ats</code>)</li>	<li>Use one of the AWS Root CAs (<a href="https://www.amazontrust.com/repository/AmazonRootCA1.pem" rel="nofollow noreferrer">https://www.amazontrust.com/repository/AmazonRootCA1.pem</a>)</li>	</ol>		<p>Full details: <a href="https://aws.amazon.com/blogs/iot/aws-iot-core-ats-endpoints/" rel="nofollow noreferrer">https://aws.amazon.com/blogs/iot/aws-iot-core-ats-endpoints/</a></p>	"
53119129,93230919,53119258,3,0,Tue Nov 06 11:05:00 EST 2018,5114014,"I am using -ats and updated the old certificate with this link that you have mentioned: https://www.amazontrust.com/repository/AmazonRootCA1.pem		Now its giving this error:	ValueError: Invalid endpoint pattern for wss: xxxxxxxx-ats.iot.us-west-2.amazonaws.com"
